client/my-sites/comments/controller.js

```diff
@@ -9,10 +9,17 @@ import { includes } from 'lodash';
 /**
  * Internal dependencies
  */
-import route from 'lib/route';
 import CommentsManagement from './main';
+import config from 'config';
+import route from 'lib/route';
 
-const VALID_STATUSES = [ 'pending', 'approved', 'spam', 'trash', 'all' ];
+const VALID_STATUSES = [
+	'pending',
+	'approved',
+	'spam',
+	'trash',
+	...config.isEnabled( 'comments/management/all-list' ) && 'all',
```

Comment:
This seems a bit clever. To make things more clear, could we switch this assignment out to a ternary, or modify the list after? 
```jsx
if ( config.isEnabled( 'comments/management/all-list' ) ) {
    VALID_STATUSES.push( 'all' );
}
```